matlab/test/arrow/array/tDate64Array.m

```diff
@@ -0,0 +1,308 @@
+            array = testCase.ArrowArrayConstructorFcn(dates);
+            testCase.verifyEqual(array.Length, int64(0));
+            testCase.verifyEqual(array.Valid, logical.empty(0, 1));
+            testCase.verifyEqual(toMATLAB(array), datetime.empty(0, 1));
+        end
+
+        function TestErrorIfNonVector(testCase)
+            % Verify arrow.array.Date64Array.fromMATLAB() throws an error
+            % if the input provided is not a vector.
+
+            dates = datetime(2023, 1, 1) + days(1:12);
+            dates = reshape(dates, 2, 6);
+            fcn = @() testCase.ArrowArrayConstructorFcn(dates);
+            testCase.verifyError(fcn, "arrow:array:InvalidShape");
+
+            dates = reshape(dates, 3, 2, 2);
+            fcn = @() testCase.ArrowArrayConstructorFcn(dates);
+            testCase.verifyError(fcn, "arrow:array:InvalidShape");
+        end
+
+        function TestErrorIfNonDatetime(testCase)
+            % Verify arrow.array.Date64Array.fromMATLAB() throws an error
+            % if not given a datetime as input.
+
+            dates = duration(1, 2, 3);
+            fcn = @() testCase.ArrowArrayConstructorFcn(dates);
+            testCase.verifyError(fcn, "arrow:array:InvalidType");
+
+            numbers = [1; 2; 3; 4];
+            fcn = @() testCase.ArrowArrayConstructorFcn(numbers);
+            testCase.verifyError(fcn, "arrow:array:InvalidType");
+        end
+
+        function TestInt64MaxMilliseconds(testCase)
+            % Verify that no precision is lost when trying to round-trip a
+            % datetime value that is within abs(intmin("int64")) milliseconds before
+            % and intmax("int64") days after the UNIX epoch.
+
+            ticksPerSecond = 1e3;
+
+            numMillisecondsBefore = intmin("int64");
+            numMillisecondsAfter = intmax("int64");
+
+            expectedBefore = datetime(numMillisecondsBefore, ConvertFrom="epochtime", ...
+                Epoch=testCase.UnixEpoch, TicksPerSecond=ticksPerSecond);
+            expectedAfter = datetime(numMillisecondsAfter, ConvertFrom="epochtime", ...
+                Epoch=testCase.UnixEpoch, TicksPerSecond=ticksPerSecond);
+
+            array = testCase.ArrowArrayConstructorFcn(expectedBefore);
+            actualBefore = array.toMATLAB();
+            testCase.verifyEqual(actualBefore, expectedBefore);
+
+            array = testCase.ArrowArrayConstructorFcn(expectedAfter);
+            actualAfter = array.toMATLAB();
+            testCase.verifyEqual(actualAfter, expectedAfter);
+        end
+
+        function TestZonedDatetime(testCase)
+            % Verify that zoned datetimes are supported as inputs to the
+            % fromMATLAB method and that the output datetime returned by
+            % the toMATLAB method is unzoned.
+            zoned = testCase.UnixEpoch + days(10);
+            zoned.TimeZone = "America/New_York";
+
+            expected = zoned;
+            expected.TimeZone = "UTC";
```

Comment:
Should we add a comment to explain why we set `TimeZone` to `"UTC"` before setting to `''`.  It may not be obvious that changing the `TimeZone` from `America/New_York` to `UTC` adjusts the value so that it represents the same "instant-in-time".

Reply:
That's a great idea.

Reply:
Fixed.